### PR TITLE
Fix tests broken by #72

### PR DESCRIPTION
**Pull request by scottbommarito**
Fixes https://github.com/NuGet/NuGet.Jobs/issues/73.

Basically, there were test cases that put a null/empty value into the configuration dictionary and expected it to throw. Since we were not using the latest implementation of the [dictionary extensions](https://github.com/NuGet/ServerCommon/blob/develop/src/NuGet.Services.Configuration/DictionaryExtensions.cs), which were a replacement for another method that had the correct behavior, the configuration accesses did not throw.

`GetOrNull<T>` was a method that returned a converted value from the dictionary or null if no value exists. It has been replaced by `GetOrDefault<T>`, a method that returns a converted value from the dictionary or a default (`default(T)` or a value specified by a parameter).

TODO:
- [x] Update with newest references to `Server.Common` packages when [this PR](https://github.com/NuGet/ServerCommon/pull/14) is done.

### CODE REVIEW

File: src/Stats.CollectAzureCdnLogs/packages.config [hunk 1]
```diff
@@ -27,8 +27,8 @@
   <package id="Microsoft.Net.Http" version="2.2.29" targetFramework="net452" />
   <package id="Microsoft.WindowsAzure.ConfigurationManager" version="3.2.1" targetFramework="net452" />
   <package id="Newtonsoft.Json" version="9.0.1" targetFramework="net452" />
-  <package id="NuGet.Services.Configuration" version="1.0.68-r-develop" targetFramework="net452" />
-  <package id="NuGet.Services.KeyVault" version="1.0.68-r-develop" targetFramework="net452" />
+  <package id="NuGet.Services.Configuration" version="1.0.110-r-sb-getorthrow" targetFramework="net452" />
+  <package id="NuGet.Services.KeyVault" version="1.0.110-r-sb-getorthrow" targetFramework="net452" />
```
Comment (shishirx34) [inline src/Stats.CollectAzureCdnLogs/packages.config]: Can we not do this? and only consume from a single place. Merge your changes to develop branch.

Comment (shishirx34) [inline src/Stats.CollectAzureCdnLogs/packages.config]: Why can't you use version: 1.0.106-r-develop

Comment (shishirx34) [inline src/Stats.CollectAzureCdnLogs/packages.config]: Ah, I see your TODO. You should get that changed merged in before sending a PR here.

**Comment by scottbommarito** [comment]
Have decided to revert #72 instead of merging this fix. When we have more unit testing we can explore this option again.

**Pull request closed without merging**